### ISSUE TITLE
Help with flakes

### ISSUE THREAD

**Issue by bivsk** [Jappie3/hyprcursor-phinger#1]
Hello, forgive the basic question, but I'd really appreciate some help getting this to work with my flake. I'm still getting introduced to flakes, and I can't manage to get these cursor themes installed.

So far, I've added the flake's URL to my inputs, and "installed" the package via:
```
home.packages = with pkgs; [
  hyprcursor
  inputs.hyprcursor-phinger.packages."x86_64-linux".hyprcursor-phinger
];
```

At this stage, I'm confused, since I cannot find where/if the cursor actually got installed or not. Running `nix build github:Jappie3/hyprcursor-phinger` seems to work, however, I don't see a `~/.local/share/icons` directory after actually including the package (as above).

Additionally, I noticed there was a hm-module included. I think I'd prefer using this over the package in my system config, but I can't seem to figure out how to use out-of-tree home-manager modules. Do I need to add this flake as a module for my home-manager config somehow? I.e: 
```
homeConfigurations = {
  ...
  modules = [
    ./home
    inputs.hyprcursor-phinger  <---???
  ];
};
```

Any help would be greatly appreciated.. I'm getting close to being able to do this on my own just having some trouble fitting all the pieces together.  Thanks :)

**Comment by Jappie3** [comment]
hm I should've probably included some instructions...

adding the flake URL to inputs is correct, but you don't 'install' a cursortheme like that

adding things to `home.packages` or `environment.systemPackages` means that the `bin/` folder in the flake output gets added to `$PATH` - this flake does not have a `result/bin/` folder since there are no binaries:

```
 tree result -L 2
result
└── cursors
    ├── theme_phinger-cursors-dark
    └── theme_phinger-cursors-light

4 directories, 0 files
```

instead, you can use the HM module to put the cursor theme at `~/.local/share/icons`:

```nix
  # first, import the HM module from this flake
  imports = [
    inputs.hyprcursor-phinger.homeManagerModules.hyprcursor-phinger
  ];
  # then, enable it:
  programs.hyprcursor-phinger.enable = true;
```

in the `hm-module.nix`, you can see this option being declared & used for putting the theme in the correct position:

```nix
  options.programs.hyprcursor-phinger.enable = lib.mkEnableOption "hyprcursor-phinger";

  config = lib.mkIf config.programs.hyprcursor-phinger.enable {
    home.file = {
      "phinger-cursors-light" = {
        source = "${self.packages.${pkgs.system}.hyprcursor-phinger}/cursors/theme_phinger-cursors-light";
        target = ".local/share/icons/phinger-cursors-light";
      };
      "phinger-cursors-dark" = {
        source = "${self.packages.${pkgs.system}.hyprcursor-phinger}/cursors/theme_phinger-cursors-dark";
        target = ".local/share/icons/phinger-cursors-dark";
      };
    };
  };
```

hope this helps, lmk if you have any more questions ^^

**Comment by Jappie3** [comment]
fixed in 52980f831b3667e49eb2717c74eb0851f104d63e

**Comment by bivsk** [comment]
Thank you so much! That clarified things nicely :)
